lua/lualine/components/tabs/tab.lua

```diff
@@ -14,6 +14,21 @@ end
 --- of the tab.
 ---@return string
 function Tab:label()
+  local custom_tabname
+  if vim.g.lualine_tabnames then
+    custom_tabname = vim.g.lualine_tabnames[tostring(self.tabnr)]
+  else
+    vim.cmd(
+    [[
+    if(exists("g:Lualine_tabnames_str"))
+      let g:lualine_tabnames = json_decode(g:Lualine_tabnames_str)
```

Comment:
I don't like this . Why are we doing this ? We should just do tabpage variables like before .

Reply:
With this, we can save tabnames with `:mksession` and restore. `mksession` can only save strings and numbers, so I have to  maintain another variable `g:Lualine_tabnames_str`.

Reply:
Note: `shada` by default will store that variable too creating a mess .

If neovim doesn't provide an way to store tabpage variables in session, open an issue there . Lualine shouldn't work workaround it in this manner . Sessions aren't concern of lualine.